openrank/apiserver/models.py

```diff
@@ -1,3 +1,126 @@
 from django.db import models
 
 # Create your models here.
+#All the Comments and doc-strings may contain Opinionated Views.
+#These are not the final docs. Constructive Negation is welcome!  ex  you may find the comments too-verbose
+
+class Contest(models.Model):
+    name=models.CharField()
+    description=models.TextField()
+    url = models.URLField()
+    is_public = models.BooleanField()
+    starttime = models.TimeField()
+    user_id = models.ForeignKey(User)
+    position = models.ForeignKey(Position)
+    cutoff_duration = models.IntegerField()
+    login_start_time = models.TimeField()
+    password = models.charField(max_length=45)
+
+
+class Position(models.models):
+	job_title = models.charField()
+
+
+
+
+
+class User(models.Model):
+	'''Currently these fiels are planned
+	1)name -eh just the name.     should there be a max length? for name? 
+	2)email - email of the contestant.  BuiltIn  EmailField from django is being used.
+	3)password-
+	4)phone_number- an IntegerField.    how will we validate this? is there any Buildin  (like how it is for Email)
+
+'''
+
+    role = models.OneToOneField()  # A Role can oprionally be assigned to a User. But a User MUST be assigned to SOME ROLE
+	name  = models.CharField(max_length=60)
+	email = models.EmailField(dmax_length=60)
+	password = models.CharField(max_length=50)
+	phone_number = models.charField(max_length=10)
+
+
+class Enrollment(models.Model)
+    user_id =    models.ForeignKey(User)
+    contest_id = models.ForeignKey(Contest,on_delete=models.CASCADE)
+    marksobtained = models.IntegerField()
```

Comment:
can we just use marks_obtained or one way of separating words?

Reply:
sure we must.